hack/make/validate-vet

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+source "$(dirname "$BASH_SOURCE")/.validate"
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only -- '*.go' | grep -v '^vendor/' || true) )
+unset IFS
+
+badFiles=()
+for f in "${files[@]}"; do
+	# we use "git show" here to validate that what's committed is vetted
+	if [ "$(git show "$VALIDATE_HEAD:$f" | go tool vet)" ]; then
+		badFiles+=( "$f" )
```

Comment:
can we record here messages from vet directly? Thus we don't need to run vet twice.